src/main/java/com/company/managementservice/controller/EmployeeController.java

```diff
@@ -0,0 +1,95 @@
+package com.company.managementservice.controller;
+
+import com.company.managementservice.exception.NotFoundException;
+import com.company.managementservice.model.dto.EmployeeDto;
+import com.company.managementservice.model.response.BaseMessageResponse;
+import com.company.managementservice.model.response.ServiceResponse;
+import com.company.managementservice.service.EmployeeService;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@Log4j2
+@RestController()
+@RequestMapping(value = "/employee", consumes = MediaType.APPLICATION_JSON_VALUE)
+public class EmployeeController {
+
+    @Autowired
+    private EmployeeService employeeService;
+
+    @PostMapping
+    public ServiceResponse<?> postEmployeeDetails(@Valid @RequestBody EmployeeDto employeeDto) {
+        log.info(
+                "EmployeeController : postEmployeeDetails : Received Request to post Employee Details" +
+                employeeDto.toString());
+        return new ServiceResponse<BaseMessageResponse>(
+                new BaseMessageResponse("Saved Successfully  " + employeeService.saveEmployee(employeeDto).toString(),
+                                        HttpStatus.OK, true));
+
+    }
+
+    @PutMapping(value = "/{employeeId}/department/{departmentId}")
+    public ServiceResponse<?> assignEmployeeToDepartment(@NonNull @PathVariable Long departmentId, @NonNull @PathVariable Long employeeId
```

Comment:
I have assumed that Organisation will have active and inactive departments , departments can have active and inactive employees, so both will be visible to client

Reply:
Assumption 


1. Organisation can exist without departments,departments can exist without employees , employee can exist without salary(unpaid intern) and departments (employee may be removed from a project in one department and waiting for assignment of project in another department)
2. Organisation will have active and inactive departments , departments can have active and inactive employees , inactive employee cannot have active salary —>2.1 department is made inactive then it will still have employees but without any worth .(Todo we can remove employees as step 4) 
3. department can be in multiple organisation, employee  can be in multiple departments 
4. departments can be removed from organisation , employee can be removed from department 
5. if dont show inactive department in organisation then do step 4 .
6. For now i have taken only one role admin for update operations 
7. employee are free to retrieve /employee/id each other details but not salary 
8. each employe has only one role